### PR TITLE
Flows by port: add condition to check switch ID and separate scr and dst ports when processing segments.

### LINKED ISSUE #5450: Incorrect flow by port information
**Steps to reproduce:**
1. Create a multi switch flow via API `POST /v2/flows`
```json
{
  "flow_id": "flow1",
  "source": {
    "inner_vlan_id": 0,
    "port_number": 10,
    "switch_id": "2",
    "vlan_id": 123
  },
  "destination": {
    "inner_vlan_id": 0,
    "port_number": 10,
    "switch_id": "9",
    "vlan_id": 456
  },
  "maximum_bandwidth": 1000
}
```

2. Get Path for the flow via API `GET /v1/flows/{flow-id}/path`

```json
{
  "flowid": "flow1",
  "flowpath_forward": [
    {
      "switch_id": "00:00:00:00:00:00:00:02",
      "input_port": 10,
      "output_port": 3
    },
    {
      "switch_id": "00:00:00:00:00:00:00:03",
      "input_port": 1,
      "output_port": 5
    },
    {
      "switch_id": "00:00:00:00:00:00:00:09",
      "input_port": 32,
      "output_port": 10
    }
  ],
  "flowpath_reverse": [
    {
      "switch_id": "00:00:00:00:00:00:00:09",
      "input_port": 10,
      "output_port": 32
    },
    {
      "switch_id": "00:00:00:00:00:00:00:03",
      "input_port": 5,
      "output_port": 1
    },
    {
      "switch_id": "00:00:00:00:00:00:00:02",
      "input_port": 3,
      "output_port": 10
    }
  ]
}
```

3. Get flows by port for src switch via API `GET /v2/switches/{switch_id}/flows-by-port`

**Expected result:**
Response contains ports 10 and 3 because flow1 uses only these ports on switch 2.

**Actual result:**
Response contains ports 10, 3 and 1 because port 1 belongs to the first flow segment sw1, port 3  <====> sw3, port1

```json
{
  "flows_by_port": {
    "1": [
      {
        "flow_id": "flow1",
        "source": {
          "switch_id": "00:00:00:00:00:00:00:02",
          "port_number": 10,
          "vlan_id": 123,
          "inner_vlan_id": 0,
          "detect_connected_devices": {
            "lldp": false,
            "arp": false
          }
        },
        "destination": {
          "switch_id": "00:00:00:00:00:00:00:09",
          "port_number": 10,
          "vlan_id": 456,
          "inner_vlan_id": 0,
          "detect_connected_devices": {
            "lldp": false,
            "arp": false
          }
        },
        "status": "Up",
        "status_details": null,
        "status_info": null,
        "maximum_bandwidth": 1000,
        "ignore_bandwidth": false,
        "strict_bandwidth": false,
        "periodic_pings": false,
        "description": null,
        "max_latency": null,
        "max_latency_tier2": null,
        "priority": null,
        "diverse_with": null,
        "diverse_with_y_flows": null,
        "diverse_with_ha_flows": null,
        "affinity_with": null,
        "pinned": false,
        "allocate_protected_path": false,
        "encapsulation_type": "transit_vlan",
        "path_computation_strategy": "cost_and_available_bandwidth",
        "target_path_computation_strategy": null,
        "loop_switch_id": null,
        "forward_path_latency_ns": 0,
        "reverse_path_latency_ns": 0,
        "latency_last_modified_time": null,
        "created": "2023-10-20T12:15:30.415Z",
        "last_updated": "2023-10-20T12:15:33.466Z",
        "mirror_point_statuses": null,
        "statistics": {
          "vlans": null
        }
      }
    ],
    "3": [
      {
        "flow_id": "flow1",
        "source": {
          "switch_id": "00:00:00:00:00:00:00:02",
          "port_number": 10,
          "vlan_id": 123,
          "inner_vlan_id": 0,
          "detect_connected_devices": {
            "lldp": false,
            "arp": false
          }
        },
        "destination": {
          "switch_id": "00:00:00:00:00:00:00:09",
          "port_number": 10,
          "vlan_id": 456,
          "inner_vlan_id": 0,
          "detect_connected_devices": {
            "lldp": false,
            "arp": false
          }
        },
        "status": "Up",
        "status_details": null,
        "status_info": null,
        "maximum_bandwidth": 1000,
        "ignore_bandwidth": false,
        "strict_bandwidth": false,
        "periodic_pings": false,
        "description": null,
        "max_latency": null,
        "max_latency_tier2": null,
        "priority": null,
        "diverse_with": null,
        "diverse_with_y_flows": null,
        "diverse_with_ha_flows": null,
        "affinity_with": null,
        "pinned": false,
        "allocate_protected_path": false,
        "encapsulation_type": "transit_vlan",
        "path_computation_strategy": "cost_and_available_bandwidth",
        "target_path_computation_strategy": null,
        "loop_switch_id": null,
        "forward_path_latency_ns": 0,
        "reverse_path_latency_ns": 0,
        "latency_last_modified_time": null,
        "created": "2023-10-20T12:15:30.415Z",
        "last_updated": "2023-10-20T12:15:33.466Z",
        "mirror_point_statuses": null,
        "statistics": {
          "vlans": null
        }
      }
    ],
    "10": [
      {
        "flow_id": "flow1",
        "source": {
          "switch_id": "00:00:00:00:00:00:00:02",
          "port_number": 10,
          "vlan_id": 123,
          "inner_vlan_id": 0,
          "detect_connected_devices": {
            "lldp": false,
            "arp": false
          }
        },
        "destination": {
          "switch_id": "00:00:00:00:00:00:00:09",
          "port_number": 10,
          "vlan_id": 456,
          "inner_vlan_id": 0,
          "detect_connected_devices": {
            "lldp": false,
            "arp": false
          }
        },
        "status": "Up",
        "status_details": null,
        "status_info": null,
        "maximum_bandwidth": 1000,
        "ignore_bandwidth": false,
        "strict_bandwidth": false,
        "periodic_pings": false,
        "description": null,
        "max_latency": null,
        "max_latency_tier2": null,
        "priority": null,
        "diverse_with": null,
        "diverse_with_y_flows": null,
        "diverse_with_ha_flows": null,
        "affinity_with": null,
        "pinned": false,
        "allocate_protected_path": false,
        "encapsulation_type": "transit_vlan",
        "path_computation_strategy": "cost_and_available_bandwidth",
        "target_path_computation_strategy": null,
        "loop_switch_id": null,
        "forward_path_latency_ns": 0,
        "reverse_path_latency_ns": 0,
        "latency_last_modified_time": null,
        "created": "2023-10-20T12:15:30.415Z",
        "last_updated": "2023-10-20T12:15:33.466Z",
        "mirror_point_statuses": null,
        "statistics": {
          "vlans": null
        }
      }
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by dmitrii-beliakov**
This change fixes the bug when redundant ports appeared in the flows by port response.

closes #5450

### CODE REVIEW

File: src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepository.java [hunk 1]
```diff
@@ -560,11 +560,13 @@ public Map<Integer, Collection<Flow>> findSwitchFlowsByPort(SwitchId switchId, C
...
                     ).getRawTraversal().toList().stream().map(Map.class::cast),
-                    PathSegmentFrame.SRC_PORT_PROPERTY, PathSegmentFrame.DST_PORT_PROPERTY);
+                    switchId);
 
+            log.error("FermaFlowRepository populate from segments: {}", portToFlowMap);
             addFlowEndPoints(switchId, portToFlowMap);
-
+            log.error("FermaFlowRepository add end points: {}", portToFlowMap);
             portToFlowMap.entrySet().removeIf(e -> (ports != null && !ports.contains(e.getKey())));
+            log.error("FermaFlowRepository remove not needed ports {}: {}", ports, portToFlowMap);
```
Comment (niksv) [inline src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepository.java:569]: Did you forget to remove debug logs?

**Review by niksv — COMMENTED** [review]
Could you please add unit tests?

File: src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepository.java (hunk 1 above)
Comment (dmitrii-beliakov) [reply src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepository.java:569]: yes, I'll remove it now

**Comment by dmitrii-beliakov** [comment]
> Could you please add unit tests?

I am thinking about how to do that. Without DB it doesn't seem useful, some integration test could catch things like this bug, but we need to spin up DB for that.

**Comment by niksv** [comment]
@dmitrii-beliakov We have `FermaFlowRepositoryTest` 
As far as I remember in runs in memory DB

**Comment by dmitrii-beliakov** [comment]
> @dmitrii-beliakov We have `FermaFlowRepositoryTest` As far as I remember in runs in memory DB

I adjusted a unit test. The test didn't catch the bug, so I first thought that it is the env issue. But it turned out that in reality it was the test data masking this bug. Just unfortunate select of values and corner cases. Now I split the cases into separate tests.

**Review by niksv — APPROVED** [review]

**Comment by pkazlenka** [comment]
Verified manually. No need to add functional autotest for this small bugfix.

**Review by IvanChupin — APPROVED** [review]
